### PR TITLE
sql/tests: fix secondary index key in sysbenchKV.DeleteInsert

### PR DESCRIPTION
**Pull request by nvb**
`sysbenchKV.DeleteInsert` was previously re-inserting the secondary index key that it had just deleted, instead of inserting the secondary index key that corresponds to the new primary index key.

To prevent this kind of bug in the future, we add some checks for secondary index integrity, where appropriate.

Epic: None
Release note: None

### CODE REVIEW

File: pkg/sql/tests/sysbench_test.go [hunk 1]
```diff
@@ -459,36 +472,26 @@ func (s *sysbenchKV) indexKey(t tableNum, k kValue, id rowID) []byte {
...
 	s.encodePKValue(b, k)
-	var v roachpb.Value
-	v.SetBytes(b)
-	v.InitChecksum(pkKey)
-	return v.RawBytes
+	return b
 }
 
 func (s *sysbenchKV) encodePKValue(b []byte, k kValue) {
 	// Replace the first 8 bytes with the k value, so that we can decode k to keep
```
Comment (tbg) [inline pkg/sql/tests/sysbench_test.go:483]: The comment is stale.

Comment (tbg) [reply pkg/sql/tests/sysbench_test.go:483]: Actually it's not, nevermind.

**Review by tbg — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/137265#-:-ODu23d49N26i4Shyrxa:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @nvanbenschoten and @RaduBerinde)

___
*[`pkg/sql/tests/sysbench_test.go` line 478 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/137265#-ODu1aXC0O1TzUVB52Ca:-ODu1aXC0O1TzUVB52Cb:b-ytww5s) ([raw file](https://github.com/cockroachdb/cockroach/blob/bfdabf4fdde5ebbd9af2f6493682ca79abaeed13/pkg/sql/tests/sysbench_test.go#L478)):*
> ```Go
> 
> func (s *sysbenchKV) decodePKValue(b []byte) kValue {
> 	// Extract the k value from the first 8 bytes. The rest is random.
> ```

Possibly also merits an update to the comment.

**Comment by nvanbenschoten** [comment]
TFTR!

bors r+

**Pull request merged**